flytekit/core/resources.py

```diff
@@ -26,24 +26,24 @@ class Resources(object):
     Also refer to the `K8s conventions. <https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#resource-units-in-kubernetes>`__
     """
 
-    cpu: Optional[str] = None
-    mem: Optional[str] = None
-    gpu: Optional[str] = None
-    storage: Optional[str] = None
-    ephemeral_storage: Optional[str] = None
+    cpu: Optional[Union[str, int]] = None
+    mem: Optional[Union[str, int]] = None
+    gpu: Optional[Union[str, int]] = None
+    storage: Optional[Union[str, int]] = None
+    ephemeral_storage: Optional[Union[str, int]] = None
 
     def __post_init__(self):
-        def _check_none_or_str(value):
+        def _check_none_or_str_or_int(value):
             if value is None:
                 return
-            if not isinstance(value, str):
-                raise AssertionError(f"{value} should be a string")
+            if not isinstance(value, (str, int)):
+                raise AssertionError(f"{value} should be a string or an integer")
 
-        _check_none_or_str(self.cpu)
-        _check_none_or_str(self.mem)
-        _check_none_or_str(self.gpu)
-        _check_none_or_str(self.storage)
-        _check_none_or_str(self.ephemeral_storage)
+        _check_none_or_str_or_int(self.cpu)
```

Comment:
LGTM. nit: could we change the name to `_check_resource_format` or something else?